test/ExportQueryExecutionTreesTest.cpp

```diff
@@ -1667,3 +1667,109 @@ TEST(ExportQueryExecutionTrees, convertGeneratorForChunkedTransfer) {
               AllOf(HasSubstr("!!!!>># An error has occurred"),
                     HasSubstr("A very strange")));
 }
+
+TEST(ExportQueryExecutionTrees, idToLiteralOrIriFunctionality) {
+  std::string kg =
+      "<s> <p> \"something\" . <s> <p> 1 . <s> <p> "
+      "\"some\"^^<http://www.w3.org/2001/XMLSchema#string> . <s> <p> "
+      "\"dadudeldu\"^^<http://www.dadudeldu.com/NoSuchDatatype> .";
+  auto qec = ad_utility::testing::getQec(kg);
+  auto getId = ad_utility::testing::makeGetId(qec->getIndex());
+  using enum Datatype;
+
+  auto checkIdToLiteralOrIri =
+      [&](Id id,
+          const std::vector<std::tuple<bool, std::optional<std::string>>>&
+              cases) {
+        for (const auto& [onlyLiteralsWithXsdString, expected] : cases) {
+          auto result = ExportQueryExecutionTrees::idToLiteral(
+              qec->getIndex(), id, LocalVocab{}, onlyLiteralsWithXsdString);
+          if (expected) {
+            EXPECT_THAT(result,
+                        ::testing::Optional(::testing::ResultOf(
+                            [](const auto& literalOrIri) {
+                              return literalOrIri.toStringRepresentation();
+                            },
+                            ::testing::StrEq(*expected))));
+          } else {
+            EXPECT_EQ(result, std::nullopt);
+          }
+        }
+      };
+
+  // Test cases: Each tuple describes one test case.
+  // The first element is the ID of the element to test.
+  // The second element is a list of 2 configurations:
+  // 1. for literals all datatypes except for xsd:string are removed, IRIs
+  // are converted to literals
+  // 2. only literals with `xsd:string` or no datatype are returned
+  std::vector<
+      std::tuple<Id, std::vector<std::tuple<bool, std::optional<std::string>>>>>
+      testCases = {
+          // Case: Literal without datatype
+          {getId("\"something\""),
+           {{false, "\"something\""}, {true, "\"something\""}}},
+
+          // Case: Literal with datatype `xsd:string`
+          {getId("\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"),
+           {{false, "\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"},
+            {true, "\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"}}},
+
+          // Case: Literal with unknown datatype
+          {getId("\"dadudeldu\"^^<http://www.dadudeldu.com/NoSuchDatatype>"),
+           {{false, "\"dadudeldu\""}, {true, std::nullopt}}},
+
+          // Case: IRI
+          {getId("<s>"), {{false, "\"s\""}, {true, std::nullopt}}},
+
+          // Case: datatype `Int`
+          {ad_utility::testing::IntId(1),
+           {{false, "\"1\""}, {true, std::nullopt}}},
+
+          // Case: Undefined ID
+          {ad_utility::testing::UndefId(),
+           {{false, std::nullopt}, {true, std::nullopt}}}};
+
+  for (const auto& [id, cases] : testCases) {
+    checkIdToLiteralOrIri(id, cases);
+  }
+}
+
```

Comment:
Add the empty comments `// _________________________________________` before all test cases for readability.